feature-libs/organization/unit-order/components/unit-level-order-history/filter/unit-level-order-history-filter.component.ts

```diff
@@ -0,0 +1,226 @@
+    } else {
+      this.renderer.setStyle(this.removeAppliedFilters.nativeElement, 'display', 'none');
+    }
+
+    this.renderer.setStyle(this.filterNav.nativeElement, 'display', 'none');
+    this.renderer.setStyle(this.filterNavUnit.nativeElement, 'display', 'none');
+    this.renderer.setStyle(this.filterNavBuyer.nativeElement, 'display', 'none');
+  }
+
+  closeFilterNav(): void {
+    this.renderer.setStyle(this.filterNav.nativeElement, 'display', 'none');
+    this.renderer.setStyle(this.filterNav.nativeElement, 'width', '0');
+    this.renderer.setStyle(document.body, 'overflow', '');
+  }
+
+  closeFilterSubNav(): void {
+    this.renderer.setStyle(this.filterNavUnit.nativeElement, 'display', 'none');
+    this.renderer.setStyle(this.filterNavBuyer.nativeElement, 'display', 'none');
+    this.renderer.setStyle(document.body, 'overflow', '');
+  }
+
+  backFilterSubNav(): void {
+    this.renderer.setStyle(this.filterNavUnit.nativeElement, 'display', 'none');
+    this.renderer.setStyle(this.filterNavBuyer.nativeElement, 'display', 'none');
+    this.renderer.setStyle(this.filterNav.nativeElement, 'display', 'flex');
+  }
+
+  launchSubNav(option: string): void {
+    this.renderer.setStyle(this.filterNav.nativeElement, 'display', 'none');
+
+    if (option === this.filterByUnit) {
+      this.renderer.setStyle(this.filterNavUnit.nativeElement, 'display', 'block');
+    } else if (option === this.filterByBuyer) {
+      this.renderer.setStyle(this.filterNavBuyer.nativeElement, 'display', 'block');
+    }
+  }
+
+  clearUnit(): void {
+    this.filterForm.get('unitFilter')?.reset();
+    this.renderer.setStyle(this.unitButton.nativeElement, 'display', 'none');
+    this.renderer.setStyle(this.unitPresentation.nativeElement, 'display', 'block');
+    this.searchUnitLevelOrders();
+  }
+
+  clearBuyer(): void {
+    this.filterForm.get('buyerFilter')?.reset();
+    this.renderer.setStyle(this.buyerButton.nativeElement, 'display', 'none');
+    this.renderer.setStyle(this.buyerPresentation.nativeElement, 'display', 'block');
+    this.searchUnitLevelOrders();
+  }
+
+  clearUnitMobile(): void {
+    this.filterFormMobile.get('unitFilterMobile')?.reset();
+    this.renderer.setStyle(this.unitButtonMobile.nativeElement, 'display', 'none');
+    this.renderer.setStyle(this.unitPresentationMobile.nativeElement, 'display', 'block');
+    this.searchUnitLevelOrdersForMobile();
+    this.renderer.setStyle(document.body, 'overflow', '');
+  }
+
+  clearBuyerMobile(): void {
+    this.filterFormMobile.get('buyerFilterMobile')!.reset();
+    this.renderer.setStyle(this.buyerButtonMobile.nativeElement, 'display', 'none');
+    this.renderer.setStyle(this.buyerPresentationMobile.nativeElement, 'display', 'block');
+    this.searchUnitLevelOrdersForMobile();
+    this.renderer.setStyle(document.body, 'overflow', '');
+  }
+
+  searchBuyer(inputElement: HTMLInputElement): void {
+    const value = inputElement.value;
+    if (!value || value === '') {
```

Comment:
Should we do instead `value.trim() === ''` instead (also for the following 3 occurrences)? Thoughts @KibriyaInamdar @MarcinGanczorz @pavel-ermolin?